plugins/3scale-backend/src/providers/config.ts

```diff
@@ -0,0 +1,50 @@
+import { readTaskScheduleDefinitionFromConfig } from '@backstage/backend-tasks';
+import { Config } from '@backstage/config';
+import { ThreeScaleConfig } from './types';
+
+export function readThreeScaleApiEntityConfigs(
+  config: Config,
+): ThreeScaleConfig[] {
+  const configs: ThreeScaleConfig[] = [];
+
+  const providerConfigs = config.getOptionalConfig(
+    'catalog.providers.threeScaleApiEntity',
+  );
+
+  if (!providerConfigs) {
+    return configs;
+  }
+
+  for (const id of providerConfigs.keys()) {
+    configs.push(
+      readThreeScaleApiEntityConfig(id, providerConfigs.getConfig(id)),
+    );
+  }
+
+  return configs;
+}
```

Comment:
It may be more efficient to do it like this:

```suggestion
  const providerConfigs = config.getOptionalConfig(
    'catalog.providers.threeScaleApiEntity',
  );

  if (!providerConfigs) {
    return [];
  }

  return providerConfigs.keys().map(id => readThreeScaleApiEntityConfig(id, providerConfigs.getConfig(id))))
}
```

Array `.push()` is many times slower than `.map()`.